x-template.js

```diff
@@ -1,31 +1,290 @@
-/** Internal implementation details for template engine. */
-class TemplateEngine {
+/** Forgiving HTML parser which leverages innerHTML. */
+class Forgiving {
```

Comment:
Just moving some things around to form a new conceptual boundary between our “parser” and our “dom manager”.